src/foam/core/reflow/cmd/Commands.js

```diff
@@ -218,15 +218,16 @@ foam.CLASS({
 
   requires: [ 'foam.core.boot.CSpec', 'foam.lang.Latch' ],
 
-  imports: [ 'AuthenticatedCSpecDAO as cSpecDAO' ],
+  imports: [ 'AuthenticatedCSpecDAO as cSpecDAO', 'auth' ],
 
   properties: [
-    [ 'description', 'Display available DAO services' ]
+    [ 'description', 'Display available DAO services', 'uploadAvailable' ]
   ],
 
   methods: [
     function execute(opt_nameQuery) {
       var self = this;
+      this.auth.check(this, 'command.read.upload').then(r => this.uploadAvailable = r );
```

Comment:
Maybe better if we just check if the upload command is in the CommandDAO.
commandDAO.find('upload').
Then we don't care if it's permission or just not installed or whatever.